includes/options/class-amp-options-rest-controller.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Rest endpoint for fetching and updating plugin options from admin screens.
+ *
+ * @package AMP
+ * @since 1.6.0
+ */
+
+use AmpProject\AmpWP\Option;
+
+/**
+ * AMP setup wizard class.
+ *
+ * @since 1.6.0
+ */
+final class AMP_Options_REST_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'amp/v1';
+		$this->rest_base = 'options';
+	}
+
+	/**
+	 * Registers all routes for the controller.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				[
+					'methods'             => WP_REST_SERVER::READABLE,
+					'callback'            => [ $this, 'get_items' ],
+					'args'                => [],
+					'permission_callback' => [ $this, 'get_items_permissions_check' ],
+				],
+				[
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => [ $this, 'update_items' ],
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+					'permission_callback' => [ $this, 'get_items_permissions_check' ],
+				],
+				'schema' => $this->get_public_item_schema(),
+			]
+		);
+	}
+
+	/**
+	 * Checks whether the current user has permission to retrieve options.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```

Comment:
The docblock and code state that this is only checked for read access, but this same method is also used to check whether the options can be written to as well.